atc/exec/try_step.go

```diff
@@ -7,20 +7,23 @@ import (
 // TryStep wraps another step, ignores its errors, and always succeeds.
 type TryStep struct {
 	step Step
+	aborted bool
 }
 
 // Try constructs a TryStep.
 func Try(step Step) Step {
-	return TryStep{
+	return &TryStep{
```

Comment:
How come we're changing this from a pointer?

Reply:
it's so the `ts.aborted = true` state persists through to the `Succeeded()` call

Reply:
Why does wouldn't it get persisted in the pointer case?

Reply:
assuming you mean non-pointer case, `Run` is called with a copy of the struct as its receiver, so changes are only local to that function's copy

Reply:
oh wow, I totally read this diff as going form ptr to value. haha don't mind me

Reply:
I was under the impression this was necessary to mutate it later on, but if I'm wrong i can definitely fix it

Reply:
no no its fine. I misread the diff and got confused for a second.